### ISSUE TITLE
Turning on the visibility of a layer using the "Space" button does not turn on the visibility of group layers

### ISSUE THREAD

**Issue by MrChebur** [qgis/QGIS#38259]
**Describe the bug**
![Bug animation](https://user-images.githubusercontent.com/20655102/90119495-13b79f80-dd73-11ea-9545-4b5ad887e480.gif)
Turning on the visibility of a layer using the "Space" button does not turn on the visibility of group layers containing this layer. Performing the same actions with a mouse - turns on group layers.

**QGIS and OS versions**

QGIS version
3.14.1-Pi
QGIS code revision
de08d6b71d
Compiled against Qt
5.11.2
Running against Qt
5.11.2
Compiled against GDAL/OGR
3.0.4
Running against GDAL/OGR
3.0.4
Compiled against GEOS
3.8.1-CAPI-1.13.3
Running against GEOS
3.8.1-CAPI-1.13.3
Compiled against SQLite
3.29.0
Running against SQLite
3.29.0
PostgreSQL Client Version
11.5
SpatiaLite Version
4.3.0
QWT Version
6.1.3
QScintilla2 Version
2.10.8
Compiled against PROJ
6.3.2
Running against PROJ
Rel. 6.3.2, May 1st, 2020
OS Version
Windows 10 (10.0)
Active python plugins
AppendFeaturesToLayer; 
DigitizingTools; 
latlontools; 
MagicWand-master; 
profiletool; 
qgsAzimuth; 
quick_map_services; 
SemiAutomaticClassificationPlugin; 
valuetool; 
db_manager; 
MetaSearch; 
processing

**Comment by uclaros** [comment]
I would say the bug is the other way round: when clicking with the mouse, all parent groups are checked without me checking them...

**Comment by MrChebur** [comment]
> I would say the bug is the other way round: when clicking with the mouse, all parent groups are checked without me checking them...

I do not agree with you. If you want to enable a layer that is in a disabled group of layers, automatic enabling of the group frees the user from additional unnecessary action.

**Comment by MrChebur** [comment]
> I would say the bug is the other way round: when clicking with the mouse, all parent groups are checked without me checking them...

@uclaros I do not agree with you. If you want to enable a layer that is in a disabled group of layers, automatic enabling of the group frees the user from additional unnecessary action.

**Comment by uclaros** [comment]
> automatic enabling of the group frees the user from additional unnecessary action.

This is only true if the user also wants to to enable the groups, which is not always the case.
There is a specific action on the layer right click context menu for that, *Ceck and all its parents*, and it should not be the default when checking layers IMHO.

**Comment by Pedro-Murteira** [comment]
Still true on QGIS 3.16.15 and 3.22.2.

**Comment by MrChebur** [comment]
A quick fix for `startup.py`, located in the `C:\Users\%USERNAME%\AppData\Roaming\QGIS\QGIS3` folder.

This code will turn on/off all parent layers even with space button.

```python
import qgis

def switch_all_parents(layerTreeNode):
    layerTreeNode.setItemVisibilityCheckedParentRecursive(layerTreeNode.itemVisibilityChecked())


qgis.core.QgsProject.instance().layerTreeRoot().visibilityChanged.connect(switch_all_parents)
```
![video](https://user-images.githubusercontent.com/20655102/213378709-ba6419b4-febd-46ba-88ff-0a63ddf70fa7.gif)

**Comment by MrChebur** [comment]
I closed it by accident.
